Initiate EML Assembly Line ecology metadata management tutorial (#4753)
galaxyproject/training-material: topics/ecology/tutorials/MetaShARK_tutorial/tutorial.md

key_points:
- This tool aims to improve FAIR quality of metadata focusing on user exeprience and automatic inferences
- Creating metadata as FAIR as possible is a must
- Be carefull of the format and standard of metadata used only EML metadata will work
tags:
  - Metadata
  - EML
  - Ecology
  - Biodiversity
  - FAIR
  - Data Paper
contributions:
    authorship:
        - yvanlebras
    editing:
        - yvanlebras
        - hexylena
    funding:
        - fnso2019
        - pndb

---

<p align="justify">This tutorial aims to teach <b>how to use functionalities of the EML Assembly Line R package to produce rich metadata</b> using the Ecological Metadata Language (EML) international metadata standard. Here, we will notably propose a concrete example on how to use <b>Galaxy Ecology tools to create, evaluate and modify EML metadata</b> content using both EML Assemby Line metadata template tabular files, easily readable and editable by humans, and XML file, devoted to machine.</p>

> <agenda-title></agenda-title>
>
> In this tutorial, we will cover:
>
> 1. TOC
> {:toc}
>
{: .agenda}

# 1] How can EML Assembly Line functionalities help producing rich metadata easily?

> <p align="justify"> A major gap when a researcher is writing metadata documents is the fact that metadata international <b>standards often use formats not really human readable</b> and/or editable as XML or JSON. To answer this issue, <b>Environmental Data Initiative</b> (insert:https://edirepository.org/) (EDI) through the <b>EML Assembly Line R package</b> propose to generate intermediate metadata template files using classical tabular text format.

Reviewer: what is 'insert'? Is it supposed to be a link? You are using a lot of acronyms, do you want to use the acronym feature?

Author: hmm not sure.... I change it and delete the "insert" use

> <br><br>

Reviewer: ```suggestion

```

these aren't needed, paragraphs are naturally having a break.

<br><br>
Finally, through the <b>MetaShARK R Shiny app</b> created by the french biodiversity data hub research infrastructure (Pôle national de données de Biodiversité (<b>PNDB</b>)), user can <b>use <b>MetaShark</b>, a graphical user interface to apply the EML Assembly Line workflow</b> and benefit from some additionnal functionnalities as:</p>

- Capacity to associate terminological resources terms coming from Bioportal ontologies to data attributes as keywords using CEDAR API
- Automatic fill in of personal information using ORCID API
- Automatic production of a data paper draft

<br>

> [!NOTE]
> What is a **Data Paper**?
> <p align="justify"> According to the [GBIF](https://www.gbif.org/data-papers) (Global Biodiversity Information Facility), 
> A data paper is a peer reviewed document describing a dataset, published in a peer reviewed journal. It takes effort to prepare, curate and describe data. 

> > Data papers provide recognition for this effort by means of a scholarly article.</p>

Reviewer: please use real GTN boxes.

>    https://zenodo.org/records/10663465/files/datafile_2.tsv
>    https://zenodo.org/records/10663465/files/LakeGeneva_phytoplankton_1974-2004.nc
>    https://zenodo.org/records/10663465/files/Present.Surface.pH.tif
>    ```
>    <img src="./Images/0_Upload_datafiles.png" width=100%>
> 6. Import shapefile related files into Galaxy using the Galaxy upload tool, then on the "composite" tab, specifying "shp" composite type, then uploading .dbf, .shp and .shx files on the dedicated spaces.
>    <img src="./Images/1_upload_shapefile.png" width=100%>

# 3] MetaShARK 🦈 : Write an EML document 

To deploy a MetaShARK app, you can go to the Galaxy tool [MetaShARK](https://ecology.usegalaxy.eu/root?tool_id=interactive_tool_metashark) and click "Execute". Then, you have to wait the launch of the app, and when ready to be used, you will see the message "There is an InteractiveTool result view available," with an hyperlink on the "Open" statement allowing you to reach the app clicking on it.

> [!WARNING]
> 🚧 Note that it is a beta version of MetaShARK R shiny app. You can thus encounter issues using it!

When oppening MetaShARK, you will have an interface looking like this :

> <img src="./Images/2_MetaShARK.png" width=100%>

Reviewer: ```suggestion
![](./Images/2_MetaShARK.png)
```

all of these miss alt text :)

<img src="./Images/10_MetaShARK_personal.png" width=100%>

Then you can add final elements as:

- An abstract (writing directly on the dedicated field or uploading a text file containing the abstract)

<img src="./Images/11_MetaShARK_abstract.png" width=100%>

- Methods (writing directly on the dedicated field or uploading a text file, can be in markdown, containing details of the methods used to create data files)

<img src="./Images/12_MetaShARK_methods.png" width=100%>

- Keywords, who can be linked to keyword thesaurus. This allows you to create "group" of keywords and/or refer to existing terms classifications as we can find in terminological resources such as ontologies or thesaurus.

<img src="./Images/13_MetaShARK_keywords.png" width=100%>

> [!TIP]
> **You can add [semantic annotations](https://eml.ecoinformatics.org/semantic-annotation-primer)!**
 
<p align="justify"> To do so, you need to reach the MetaShARK parameters (upper right icon) then enter your CEDAR token. To create a CEDAR account, you can register here http://cedar.metadatacenter.org/.</p>

 API key format to enter is something like: 
```

> api 205b1e521f2eaf0ad4a361c438b63205b1e521f2eaf0ad4a361c438b63c438b63

Reviewer: i checked that it isn't a live api key :) and in the process found this could use more documentation.

1. you need to register
2. and then go to the 'profile' on cedar.metadatacenter.org
3. and there you can find the API key (I was having issues locating it, otherwise.)

Author: Done, thank you!

<img src="./Images/17_MetaShARK_annotations_predicate.png" width=100%>

<img src="./Images/18_MetaShARK_annotations_object.png" width=100%>

<p align="justify">Finally, you can specify a temporal coverage and go to the last step of this MetaShARK workflow: Generat an EML metadata file! If everything is ok, you will have creation of an EML metadata file.</p>

<img src="./Images/14_MetaShARK_makeeml.png" width=100%>

<p align="justify">Once EML written, you can <b>download the data package through the button</b> "Download Data Package". This will allow you to <b>download a zip archive</b> you can unzip on your local computer. Resulting files are organized through 2 main folders : </p>

- **A main folder with data_objects**
  -  `all datafiles ` you uploaded into MetaShARK
  -  `eml ` which is the EML metadata file written in XML format
  -  `metadata_templates ` with all metadata files written in text format, column separated by tabulations
  
- A **second folder called "emldown"** where a **draft of data paper** written in html format can be accessed

<hr>
<p align="center"><b>👏 Congratulations! You've just produced your first EML yourself!👏</b></p>
<hr>

> > [!CAUTION]
> > **At this final stage, MetaShARK can freeze and show like a "front grey filter"**
> > 
> > If this is the case, you can refresh MetaShARK app pressing F5 button of your keyboard for example. To dowload the resutling data package, you can go to the "Fill In" module, select the data package you created and click on "Download.zip".
> >
> > Returning on your MetaShARK instance</tip-title>
> > If you close MetaShARK navigator window, you can reopen it clicking on Menu > User > Active InteractiveTools.
> > You can here click on "metashark visualization" hyperlink. You can then go on the Fill In module, select 
> > the data package you created and click **Load**.
> >

Reviewer: ```suggestion
> <warning-title>MetaShARK can freeze and show like a "front grey filter"</warning-title>
> 
> If this is the case, you can refresh MetaShARK app pressing F5 button of your keyboard for example. To dowload the resutling data package, you can go to the "Fill In" module, select the data package you created and click on "Download.zip".
>
> Returning on your MetaShARK instance</tip-title>
> If you close MetaShARK navigator window, you can reopen it clicking on Menu > User > Active InteractiveTools.
> You can here click on "metashark visualization" hyperlink. You can then go on the Fill In module, select 
> the data package you created and click **Load**.
{: .warning}
```

- You can at the top of the page **download the draft** in either an _HTML format_ .... :

<img src="./Images/Download_HTML.png" width=50%>

- .... or an editable _docx format_ :

<img src="./Images/Download_docx.png" width=50%>

<hr>
Voir comment mettre les deux photos côtés à côtes ou un truc mieux
<hr>

# 6] FAIR Quality Assessment report 📊✅

[FAIR](https://www.go-fair.org/fair-principles/) stand for **Findable, Accessible, Interoperable, Reusable**. 
<br> <br>

<img src="./Images/FAIR_data_principles.jpg" width=65%>

 <br>

<p align="justify">These principles were to <b>improve the access and usabiliy of data</b> by the machine and to help <b>making data reusable and shareable for users</b>. It covers the whole concept of why it is necessary to produce a rich and described metadata in order to permit external users to <b>understand and reuse data</b> for their own studies.
  
There are several ways of computing the FAIR index, for each letter of the word is associated with a degree of FAIRitude of the data.</p>

> [!NOTE]

> > The aim is to propose a metadata quality index and to see how it is possible to improve, not to punish the user 👍

Reviewer: please use proper boxes

<hr>
> Mettre une photo avec encadré rouge pour montrer où cliquer!!!⚠️

> <hr>

Reviewer: should this be translated for non-bilingual audiences?

Author: of course, sorry....

>
>    To do so, you can go on the "visualize" functionnality of each datafile, clicking on the name of the dataset on your history then on the "visualize" button at the bottom of the dataset description. you can then select "Editor / Manually edit text" to update the content and generate a new version of the dataset.
>
> 3. Rename each datafile as original names ("abstract" and "attributes_Present.Surface.pH.txt"), you can add a "modified" tag / label so you can better remember in the future the modification state.
>
> 4. Recreate a data collection with all metadata template files, taking the new "abstract" and "attributes_Present.Surface.pH.txt" files instead of old ones.
>
> 5. You can recreate an EML metadata file with [**Make EML**](https://ecology.usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/ecology/makeeml/makeeml/0.1.1+galaxy1) EAL tool and then redeploy a metashrimps tool on the new EML.
>

# 7] Conclusion

Here is the end of this short tutorial aiming in explaining the purpose of MetaShARK and how to use it.

Resumé rapide de ce qui a été vu 

1. _lorem ipsum_
2. _lorem ipsum_
3. _lorem ipsum_
4. _lorem ipsum_

> 

Reviewer: I guess this can go? or
